### PR TITLE
SSCS-10773: Update HearingsChannelMapping getHearingChannel condition…

### PR DESCRIPTION
**Pull request by ikirsanov**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10773

### Change description ###
Update HearingsChannelMapping getHearingChannel conditions, getAllHearingChannelPreferences add last parameter 'Override fields'.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsChannelMapping.java [hunk 1]
```diff
@@ -58,19 +59,21 @@ public static List<HearingChannel> getAllHearingChannelPreferences(@Valid SscsCa
...
         hearingChannels.add(individualPreferredHearingChannel);
 
         if (nonNull(caseData.getOtherParties())) {
             hearingChannels.addAll(caseData.getOtherParties().stream()
-                .map(CcdValue::getValue)
-                .map(otherParty -> getIndividualPreferredHearingChannel(
-                    otherParty.getHearingSubtype(),
-                    otherParty.getHearingOptions(), null))
-                .collect(Collectors.toList()));
+                                       .map(CcdValue::getValue)
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsChannelMapping.java:71]: Veery deep indent

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]
Are no tests breaking? Should add a couple of new ones

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsChannelMapping.java (hunk 1 above)
Comment (ikirsanov) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsChannelMapping.java]: fixed

**Review by dattipoeHMCTS — DISMISSED** [review]

**Review by dattipoeHMCTS — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsChannelMapping.java [hunk 2]
```diff
@@ -36,7 +36,8 @@ public static List<HearingChannel> getHearingChannels(@Valid SscsCaseData caseDa
     }
 
     public static HearingChannel getHearingChannel(@Valid SscsCaseData caseData) {
-        if (HearingsDetailsMapping.isPoOfficerAttending(caseData)) {
+        if (HearingsDetailsMapping.isPoOfficerAttending(caseData)
+            && isNull(OverridesMapping.getOverrideFields(caseData).getAppellantHearingChannel())) {
```
Comment (ss-evoco) [inline src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsChannelMapping.java:40]: If the override is there, dont you need to return that ?

**Review by ss-evoco — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsChannelMapping.java (hunk 2 above)
Comment (ikirsanov) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsChannelMapping.java:40]: I think that it is returning Hearing channel according to situation? I mean on the 58-63 lines it is constructing everything from OverrideFields, or Im wrong?

Comment (TeaWithLucas) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsChannelMapping.java:40]: The override is for the appellant, not the case, so you don't need this.

**Review by TeaWithLucas — CHANGES_REQUESTED** [review]

**Review by TeaWithLucas — APPROVED** [review]

**Review by dattipoeHMCTS — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsChannelMapping.java (hunk 2 above)
Comment (ikirsanov) [reply src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsChannelMapping.java]: yes it is removed

**Review by ss-evoco — APPROVED** [review]

**Pull request merged**